docs/CircleCI.md

```diff
@@ -0,0 +1,66 @@
+# CircleCI
+
+The Buildkite Migration tool's currently supported, partially supported and unsupported properties in translation CircleCI pipelines to Buildkite pipelines are listed below:
+
+## Supported properties
+
+### aliases
+
+An `alias` is a definition of a reusable snippet of a CircleCI pipeline. Defined with a `&` (anchor) - these are then substituted into the required section of the pipeline with `*`: for example, `*tests`.
+
+The Buildkite Migration tool supports reading configuration defined from anchors and utilised through the use of an `alias`.
+
+### commands
+
+A `command` defined in a CircleCI pipeline is a reusable set of instructions with parameters that can be inserted into required `job` executions. Commands can have their own set list of `steps` that are translated through to the generated [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands`. If a `command` contains a `parameters` key, its default parameters that are specified within the block is passed through into the specific command if it is utilised.
+
+### workflows
+
+A CircleCI `workflow` is a collection of `jobs`, whose ordering defines how a CircleCI pipeline is run. Workflows can be specified to run its collection of `jobs` sequentially, or in parallel, and allows for specifying dependencies (requirements) between each `workflow` that is configured. 
+
+Each `job` that is defined as part of a `workflow` within a CircleCI pipeline will be transitioned to a Buildkite [command step](https://buildkite.com/docs/pipelines/command-step) within the generated pipeline.
+
+## Partially supported properties 
+
+### jobs
+
+CircleCI `jobs` are a collection of steps that are run on a single worker unit: whether that is on a host directly, or on a virtualised host (such as within a Docker container). 
+
+The Buildkite Migration tool currently supports the following keys relative to CircleCI `jobs`:
+
+#### General
+
+- `environment`: The `job` level environment variables of a CircleCI pipeline. Applied in the generated [command step](https://buildkite.com/docs/pipelines/command-step) as [step level](https://buildkite.com/docs/pipelines/environment-variables#runtime-variable-interpolation) environment variables with the `env` key.
+- `parameters`: Reusable keys that are used within `step` definitions within a `job`. Default parameters that are specified in a `parameters` block are passed through into the [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands` if specified.
+- `steps`: A collection of commands that are executed as part of a CircleCI `job`. Steps can be defined within an `alias`.
```

Comment:
You should also create a section here with CircleCI builtin steps as there are a few that are also not supported/translatable

Reply:
Added the others - specifically ones like `shell` (a Buildkite agent start option), `parallelism` (a option we support in steps), `branches` (workflow support for build conditionals) etc